docs/guides/proxysql/concepts/opsrequest/index.md

```diff
@@ -0,0 +1,272 @@
+section_menu_id: guides
+---
+
+> New to KubeDB? Please start [here](/docs/README.md).
+
+{{< notice type="warning" message="This is an Enterprise-only feature. Please install [KubeDB Enterprise Edition](/docs/setup/install/enterprise.md) to try this feature." >}}
+
+# ProxySQLOpsRequest
+
+## What is ProxySQLOpsRequest
+
+`ProxySQLOpsRequest` is a Kubernetes `Custom Resource Definitions` (CRD). It provides a declarative configuration for [ProxySQL](https://www.proxysql.com/) administrative operations like database version upgrading, horizontal scaling, vertical scaling,reconfiguration etc. in a Kubernetes native way.
+
+## ProxySQLOpsRequest CRD Specifications
+
+Like any official Kubernetes resource, a `ProxySQLOpsRequest` has `TypeMeta`, `ObjectMeta`, `Spec` and `Status` sections.
+
+Here, some sample `ProxySQLOpsRequest` CRs for different administrative operations is given below:
+
+**Sample ProxySQLOpsRequest for upgrading database:**
+
+```yaml
+apiVersion: ops.kubedb.com/v1alpha1
+kind: ProxySQLOpsRequest
+metadata:
+  name: proxyops-upgrade
+  namespace: demo
+spec:
+  type: UpdateVersion
+  proxyRef:
+    name: proxy-server
+  upgrade:
+    targetVersion: "2.4.4-debian"
+```
+
+**Sample ProxySQLOpsRequest Objects for Horizontal Scaling of proxysql cluster:**
+
+```yaml
+apiVersion: ops.kubedb.com/v1alpha1
+kind: ProxySQLOpsRequest
+metadata:
+  name: scale-up
+  namespace: demo
+spec:
+  type: HorizontalScaling
+  proxyRef:
+    name: proxy-server
+  horizontalScaling:
+    member: 5
+
+```
+
+**Sample ProxySQLOpsRequest Objects for Vertical Scaling of the proxysql cluster:**
+
+```yaml
+apiVersion: ops.kubedb.com/v1alpha1
+kind: ProxySQLOpsRequest
+metadata:
+  name: proxyops-vscale
+  namespace: demo
+spec:
+  type: VerticalScaling
+  proxyRef:
+    name: proxy-server
+  verticalScaling:
+    proxysql:
+      requests:
+        memory: "1.2Gi"
+        cpu: "0.6"
+      limits:
+        memory: "1.2Gi"
+        cpu: "0.6"
+```
+
+**Sample ProxySQLOpsRequest Objects for Reconfiguring ProxySQL cluster:**
+
+```yaml
+apiVersion: ops.kubedb.com/v1alpha1
+kind: ProxySQLOpsRequest
+metadata:
+  name: reconfigure-vars
+  namespace: demo
+spec:
+  type: Reconfigure  
+  proxyRef:
+    name: proxy-server
+  configuration:
+    adminVariables:
+      refresh_interval: 2055
+      cluster_check_interval_ms: 205
+    mysqlVariables:
+      max_transaction_time: 1540000
+      max_stmts_per_connection: 19
+```
+
+**Sample ProxySQLOpsRequest Objects for Reconfiguring TLS of the ProxySQL:**
+
+```yaml
+apiVersion: ops.kubedb.com/v1alpha1
+kind: ProxySQLOpsRequest
+metadata:
+  name: recon-tls-update
+  namespace: demo
+spec:
+  type: ReconfigureTLS
+  proxyRef:
+    name: proxy-server
+  tls:
+    certificates:
+    - alias: server
+      subject:
+        organizations:
+        - kubedb:server
+      dnsNames:
+      - localhost
+      ipAddresses:
+      - "127.0.0.1"
+      emailAddresses:
+      - "md.alifbiswas@gmail.com"
```

Comment:
use dummy email address, not mine. :crying_cat_face: 

Applicable for other occurrences.

Reply:
resolved